### DISCUSSION TITLE
[LESS] Basic how-to guidance with Fomantic-UI-LESS?

### DISCUSSION THREAD

**Discussion by designosis** [fomantic/Fomantic-UI#1309] [General] [answered]
I'm working on a site whose users will choose from multiple themes. I've used Fomantic UI's "Default" theme as a base for all of them, modifying only a few dozen variables. So ideally, I want a bunch of files like  ...

**theme_1.less:**
```less
@import "../node_modules/fomantic-ui-less/semantic.less";
@import "my_global_style_changes.less";
// my theme 
@fontSize          : 15px;
@pageBackground    : #121212;
@primaryColor      : #f00;
// ... etc., maybe 20 more variables.
```
... so I can use gulp-less to watch and compile them for development.

Obviously the above doesn't work, and I could use help figuring out how to `@import` semantic.less and the default theme. I've renamed `node_modules/fomantic-ui-less/theme.config.example` to `theme.config`. Didn't see anything in there I needed to edit.

The above spits out `variable @red is undefined`. When `@import "../node_modules/fomantic-ui-less/theme.config";` is added, I get `variable @type is undefined in file /.../theme.less`. 

How do I add Fomantic UI's LESS as an `@import` into my own LESS file, like **theme_1.less** above?

(I'd also propose adding a line or two on "basic use" to `README.md`.)

**Comment by lubber-de** [comment, marked as answer]
The fomantic less theming system expects different themes under its `themes` directory as each component has its own set of variables

Let's say you name your theme "neokio":
- Create an own theme folder in `/fomantic-ui-less/themes/neokio`

According to your needs create specific `.variables` files in the same folder structure as in the `default` theme folder. Suppose you want to change the global font size:
- Create the file `/fomantic-ui-less/themes/neokio/globals/site.variables`
- Put `@fontSize : 15px;` into that file

Do so with every change you want to make. 
Finally set your themes name into your `/fomantic-ui-less/theme.config`:
Every component you want to use your own theme: change from `default` to `neokio`. For example

```less
/* Global */
@site       : 'neokio';
```

**Comment by designosis** [comment]
Let's say I've done the above.
Then what? How do I compile it?

**Comment by lubber-de** [comment]
It's pure LESS, so ..either use node and lessc or import lessjs if used in the browser

**Comment by designosis** [comment]
Thank you for your help! I think I have it working now.

Is it not a problem that the default `theme.config` sets the **Path to site override folder** to
`@siteFolder  : 'site';` even though the folder is actually `/node_modules/fomantic-ui-less/_site`?

**Comment by lubber-de** [comment]
Yes, this is "hidden" in the https://github.com/fomantic/Fomantic-UI/blob/master/src/README.md
When using LESS standalone without the fomantic build process (which creates the `site`folder automagically) you should rename _site to site manually

![image](https://user-images.githubusercontent.com/18379884/74043656-143d3b00-49ca-11ea-9fc3-d7ed529a9641.png)
